src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderSearchTeamCodeListAPIIntegrationTest.kt

```diff
@@ -0,0 +1,258 @@
+    }
+
+    @Test
+    fun `not allowed to do a search without COMMUNITY role`() {
+        RestAssured.given()
+                .auth()
+                .oauth2(jwtAuthenticationHelper.createJwt("ROLE_BINGO"))
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .body("""["N01000","N03000"]""")
+                .post("/team-codes")
+                .then()
+                .statusCode(403)
+    }
+
+    @Test
+    fun teamCodeListSearch() {
+        val results = RestAssured.given()
+                .auth()
+                .oauth2(jwtAuthenticationHelper.createJwt("ROLE_COMMUNITY"))
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .queryParam("page", 0)
+                .body("""["N01000"]""")
+                .post("/team-codes")
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .`as`(Array<OffenderDetail>::class.java)
+
+        Assertions.assertThat(results).hasSize(2)
+        Assertions.assertThat(results[0].offenderManagers?.get(0)?.team?.code).contains("N01000")
+        Assertions.assertThat(results[1].offenderManagers?.get(0)?.team?.code).contains("N01000")
+    }
+
+    @Test
+    fun ignoreNonActiveOffenderManagerResults() {
+        val results = RestAssured.given()
+                .auth()
+                .oauth2(jwtAuthenticationHelper.createJwt("ROLE_COMMUNITY"))
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .queryParam("page", 0)
+                .body("""["N01000","N03000"]""")
+                .post("/team-codes")
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .`as`(Array<OffenderDetail>::class.java)
+
+        Assertions.assertThat(results).hasSize(2)
+        Assertions.assertThat(results[0].offenderManagers?.get(0)?.team?.code).contains("N01000")
+        Assertions.assertThat(results[1].offenderManagers?.get(0)?.team?.code).contains("N01000")
+    }
+
+    @Test
+    fun teamCodeListSearch_ignoreNotFoundIds() {
+        val results = RestAssured.given()
+                .auth()
+                .oauth2(jwtAuthenticationHelper.createJwt("ROLE_COMMUNITY"))
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .queryParam("page", 0)
+                .body("""["N01000", "N02000", "AAAA123"]""")
+                .post("/team-codes")
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .`as`(Array<OffenderDetail>::class.java)
+
+        Assertions.assertThat(results).hasSize(3)
+        Assertions.assertThat(results[0].offenderManagers?.get(0)?.team?.code).contains("N02000")
+        Assertions.assertThat(results[1].offenderManagers?.get(0)?.team?.code).contains("N01000")
+        Assertions.assertThat(results[1].offenderManagers?.get(0)?.team?.code).contains("N01000")
```

Comment:
extra line here can be removed

Reply:
whoops, no sorry, should be checking results[2]?

Reply:
It should!